src/interceptors/XMLHttpRequest/XMLHttpRequestOverride.ts

```diff
@@ -682,5 +609,88 @@ export const createXMLHttpRequestOverride = (
         request.setRequestHeader(headerName, headerValue)
       }
     }
+
+    toPassthroughRequest(): XMLHttpRequest {
+      const request = new XMLHttpRequest()
+
+      request.open(
+        this.method,
+        this.url,
+        this.async ?? true,
+        this.user,
+        this.password
+      )
+
+      request.addEventListener('readystatechange', () => {
+        // Set original response headers as soon as they are received.
+        if (request.readyState === XMLHttpRequest.HEADERS_RECEIVED) {
+          const responseHeaders = request.getAllResponseHeaders()
+          this.log('original response headers (raw):\n', responseHeaders)
+
+          this._responseHeaders = stringToHeaders(responseHeaders)
+
+          this.log(
+            'original response headers (normalized)',
+            this._responseHeaders
+          )
+        }
+      })
+
+      request.addEventListener('progress', () => {
+        const chunk = request.response
+        const nextChunk =
+          chunk instanceof ArrayBuffer
+            ? new Uint8Array(chunk)
+            : encodeBuffer(chunk)
+
+        this._responseBuffer = concatArrayBuffer(
+          this._responseBuffer,
+          nextChunk
+        )
+      })
+
+      request.addEventListener('loadend', () => {
+        this.log('original request "loadend"')
+
+        this.status = request.status
+        this.statusText = request.statusText
+        this.responseURL = request.responseURL
+
+        this.setReadyState(this.DONE)
+        this.log(
+          'original request DONE!',
+          this.status,
+          this.statusText,
+          this.response
+        )
+      })
+
+      // Forward request headers.
```

Comment:
Since we're no longer using `propagateHeaders()`, we could maybe delete the function definition? WDYT?

Reply:
A good idea.

Reply:
Removed.